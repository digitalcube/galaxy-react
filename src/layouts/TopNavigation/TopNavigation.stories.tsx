import React from 'react';
import { Meta, Story } from '@storybook/react';
import { TopNavigation, TopNavigationProps } from './TopNavigation'

const meta: Meta = {
  title: 'Layouts/TopNavigation',
  component: TopNavigation,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<TopNavigationProps> = args => <TopNavigation {...args} />;

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
    serviceName: 'Shifter',
    today: new Date('2010-01-01')
}