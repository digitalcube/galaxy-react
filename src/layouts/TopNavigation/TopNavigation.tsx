import React, {FC} from 'react'
import { FaUser, FaSignOutAlt } from 'react-icons/fa'
import { SearchForm } from '../../components/SearchForm/SearchForm'
import { NavLinkItem, Nav, NavbarBrand, NavbarDropdown, NavbarDropdownItem } from '../../routings'

export type TopNavigationProps = {}


export const TopNavigation: FC<TopNavigationProps> = () => {
    return (
        <nav className="navbar navbar-expand-lg">
            <NavbarBrand href="/" name="shifter" />
            <SearchForm onChange={() => undefined} onSubmit={e => e.preventDefault()}/>
            <Nav>
                    <NavLinkItem active to="#">Sites</NavLinkItem>
                    <NavLinkItem to="#">Teams</NavLinkItem>
                    <NavLinkItem to="#">Guides</NavLinkItem>
                    <NavbarDropdown
                        avatar={{
                            size: 32,
                            email: "test@example.com",
                            rounded: true
                        }}
                    >
                        <NavbarDropdownItem href="#">
                            Account<FaUser />
                        </NavbarDropdownItem>
                        <NavbarDropdownItem href="#">
                            Log out <FaSignOutAlt />
                        </NavbarDropdownItem>
                    </NavbarDropdown>
                </Nav>
        </nav>
    )
}