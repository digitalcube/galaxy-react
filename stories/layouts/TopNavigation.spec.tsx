import React from 'react';
import { render } from '@testing-library/react';
import { Default as TopNavigation } from './TopNavigation.stories';

describe('Mockup/TopNavigation', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(<TopNavigation />);
    expect(asFragment()).toMatchSnapshot();
  });
});
