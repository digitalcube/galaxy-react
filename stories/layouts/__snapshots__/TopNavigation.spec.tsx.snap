// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mockup/TopNavigation to match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-light"
  >
    <a
      class="navbar-brand"
      href="/"
      to="/"
    >
      <img
        alt="shifter"
        height="32"
        width="24"
      />
    </a>
    <form
      class="form-inline my-2 my-lg-0 flex-grow-1 d-flex position-relative form-search"
    >
      <div
        class="w-100 mr-5 input-group"
      >
        <svg
          class="fas fa-search form-search-icon position-absolute"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
        <input
          aria-label="Search"
          class="form-control"
          placeholder="search"
          type="text"
          value=""
        />
      </div>
    </form>
    <button
      aria-label="Toggle navigation"
      class="navbar-toggler"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse flex-grow-0 collapse"
    >
      <ul
        class="ml-auto navbar-nav nav"
      >
        <li
          class="active nav-item"
        >
          <a
            class="rounded font-weight-bold nav-link"
            to="#"
          >
            Sites
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="rounded font-weight-bold nav-link"
            to="#"
          >
            Teams
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="rounded font-weight-bold nav-link"
            to="#"
          >
            Guides
          </a>
        </li>
        <li
          class="dropdown nav-item"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="nav-link font-weight-bold dropdown-toggle nav-link"
            data-toggle="dropdown"
            href="#"
            role="button"
          >
            <img
              alt="Gravatar for test@example.com"
              class="react-gravatar p-0  rounded-circle"
              height="32"
              src="//www.gravatar.com/avatar/55502f40dc8b7c769880b10874abc9d0?d=retro&r=g&s=32"
              srcset="//www.gravatar.com/avatar/55502f40dc8b7c769880b10874abc9d0?d=retro&r=g&s=64 2x"
              width="32"
            />
          </a>
          <div
            aria-hidden="true"
            class="dropdown-menu dropdown-menu-right"
            role="menu"
            tabindex="-1"
          >
            <button
              class="dropdown-item"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <a
                class="dropdown-item px-3 py-2 d-flex align-items-center justify-content-between"
                href="#"
                to="#"
              >
                Account
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                  />
                </svg>
              </a>
            </button>
            <button
              class="dropdown-item"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <a
                class="dropdown-item px-3 py-2 d-flex align-items-center justify-content-between"
                href="#"
                to="#"
              >
                Log out 
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
                  />
                </svg>
              </a>
            </button>
          </div>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
